app/models/article.rb

```diff
@@ -14,6 +14,15 @@ def self.create_articles(articles_data)
     end
   end
 
+  def keyword_base_text
+    base_text_data = [title,
+                      desc_facet.gsub(/,/, " "),
+                      abstract]
+    base_text_data.map do |attribute|
+      attribute.gsub(/[^\w]+/, " ").split(" ")
```

Comment:
I may be able to get rid of the first gsub and not have any issues.  I'll give that a try.